lib/index.js

```diff
@@ -14,73 +16,121 @@ function findHost() {
   return app;
 }
 
+// The root host.
+function findRoot(current) {
+  var app;
+
+  // Keep iterating upward until we don't have a grandparent.
+  // Has to do this grandparent check because at some point we hit the project.
+  // Stop at lazy engine boundaries.
+  do {
+    app = current.app || app;
+  } while (current.parent && current.parent.parent && (current = current.parent));
+
+  return app;
+}
+
+// The thing which browserify should run on.
+function findTarget(current) {
+  // If we are the project, or the project's child.
+  if (!current.parent || !current.parent.parent) {
+    return current.app;
+  } else {
+    return current.parent;
+  }
+}
+
+var Funnel = require('broccoli-funnel');
+
+function getPreprocessor(instance) {
+  return {
+    name: 'ember-browserify',
+    ext: 'js',
+    toTree: function(tree) {
+      var type = 'js';
+      var root = findRoot(instance);
+      var target = findTarget(instance);
```

Comment:
I believe this is already via the add-on API, we should use that reference to it instead.